.gitlab-ci.yml

```diff
@@ -1,31 +1,49 @@
-before_script:
-  # check the availability of some commands:
-  - command >/dev/null -v python3 jq
-
-after_script:
-  - tools/docker/stop.sh -k -r
-
 build-in-docker:
   stage: build
+  image:
+    name: prplfoundationinc/prplmesh-builder:ubuntu-18.04
+    entrypoint: [""]
+  variables:
+    CMAKE_BUILD_TYPE: "Release"
   script:
-    - cd tools/docker
-    # Try to pull from dockerhub, build the container images if that fails
-    - ./image-pull.sh || ./image-build.sh
     # Run clang-format and check there are no modified files by checking git status
     - ./clang-format.sh
     - git diff-index --exit-code HEAD
-    - ./build.sh map -f CMAKE_BUILD_TYPE=Debug BUILD_TESTS=ON
+    - cmake -DBUILD_TESTS=ON -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE -DCMAKE_INSTALL_PREFIX=build/install -H. -Bbuild -G Ninja
+    - ninja -C build install
     # Check that the AutoGenerated files are correct by checking git status
     - git update-index -q --refresh
     - git diff-index --exit-code HEAD
+
+  artifacts:
+    paths:
+      - build
```

Comment:
Do we really need to save the entire build directory? Isn't build/install sufficient?

Reply:
Done.